perma_web/frontend/components/CreateLink.vue

```diff
@@ -169,30 +118,28 @@ const handleProgressUpdate = async () => {
         clearInterval(progressInterval)
 
         captureStatus.value = 'captureError'
-        globalStore.captureErrorMessage = errorMessage
+        captureErrorMessage.value = errorMessage
     }
 }
 
-watch(() => globalStore.captureGUID, () => {
-    if (globalStore.captureErrorMessage === 'testing') {
-        return
-    }
-
-    handleProgressUpdate()
-    progressInterval = setInterval(handleProgressUpdate, 2000);
-})
-
 onMounted(() => {
+    globalStore.components.createLink = getCurrentInstance().exposed
```

Comment:
I haven't seen this pattern before. Is it a way to let components communicate with each other?

Reply:
Yeah I made this one up, but the thought process was ...

* Vue components have this nice way to encapsulate inputs and outs -- `defineProps` shows you what settings the parent component can set and two-way data bindings, and `defineExpose` shows you what methods the parent can poke. (globalState pokes holes in that whole encapsulation, so we always use that by way of a necessary or convenient evil.)
* So for the direct parent, communication is easy -- you embed `<Child ref=child props...>`, and use `child.method()` to poke child.
* Oh but then it turns out some extended relative needs to poke child sometimes too. So parent will store it as `globalState.childRef` and the relative can do `globalState.childRef.method()`. Alternatively we can have relative update a globalState variable and child watch it, which gets encapsulated in "event bus" libraries -- but _if_ we're only going to have one instance of Child, that feels like unnecessarily poking holes in Child's defineProps/defineExpose interface.
* Then after repeating this a few times, we have globalState.childRef and globalState.kidInstance and globalState.widget and it starts to feel messy, so what if we tidy them all up under a globalState.components dictionary and have the children register themselves instead of making their parents do it?

I think all of this makes sense and ends up flowing pretty naturally, _except_ that it's not really proper for the children to register themselves, because the parents are the ones in a position to know how many instances there are. So it'd probably be better to do `const childRef = ref(); globalState.components.child = childRef` in the parent. I can switch to that if you think it's worth updating.

Reply:
(This would also have the virtue that `globalState.components.child = childRef` is easier to read than `    globalStore.components.me = getCurrentInstance().exposed`)

Reply:
Interesting! Thanks for taking the time to walk me through it. It does seem like a tidy alternative to "we can have relative update a globalState variable and child watch it", if maybe a bit idiosyncratic? But no objections here! No opinion on switching things so the parent registers the child/children or not; I think I understand your argument that having the parent do it is more correct; I could believe this way is easier to read, maybe?

Reply:
`globalState.components.child = childRef` does look MUCH less fancy than `globalStore.components.me = getCurrentInstance().exposed` 🙂